### PR TITLE
[Feature-8024][Document] Add example and notice about task type Flink

### PR DESCRIPTION
**Pull request by QuakeWang**
close: https://github.com/apache/dolphinscheduler/issues/8024

I have updated the document of the Flink task, please review it. If there is nothing need to modify, I will translate and update the other relevant files.

PTAL @zhongjiajie

### CODE REVIEW

File: docs/zh-cn/2.0.3/user_doc/guide/task/flink.md [hunk 1]
```diff
@@ -1,23 +1,63 @@
...
+-    部署方式：支持 cluster 和 local 两种模式的部署。
+-    Flink 版本：根据所需环境选择对应的版本即可。
+-    任务名称（选填）：Flink 程序的名称。
+-    jobManager 内存数：用于设置 jobManager 内存数，可根据实际生产环境设置对应的内存数。
+-    Slot 数量：用于设置 Slot 的数量，可根据实际生产环境设置对应的数量。
+-    taskManager 内存数：用于设置 taskManager 内存数，可根据实际生产环境设置对应的内存数。
+-    taskManager 数量：用于设置 taskManager 的数量，可根据实际生产环境设置对应的数量。
+-    并行度：用于设置执行 Flink 任务的并行度。
+-    主程序参数：设置 Flink 程序的输入参数，支持自定义参数变量的替换。
+-    选项参数：支持 --jars、--files、--archives、--conf格式。
```
Comment (zhongjiajie) [inline docs/zh-cn/2.0.3/user_doc/guide/task/flink.md:36]: and code inline syntax
```suggestion
- 选项参数：支持 `--jars`、`--files`、`--archives`、`--conf` 格式。
```

**Review by zhongjiajie — COMMENTED** [review]
I add some suggestions, others LTGM

**Comment by zhongjiajie** [comment]
Please remember to add English version and change dev version too.

**Comment by QuakeWang** [comment]
> Please remember to add English version and change dev version too.

I have completed it.

File: docs/en-us/2.0.3/user_doc/guide/task/flink.md [hunk 2]
```diff
@@ -1,23 +1,65 @@
...
+
+## Task Example
+
+### Execute the WordCount program
+
+This is a common introductory case in the Big Data ecosystem, which often applied to computational frameworks such as MapReduce, Flink and Spark. The main purpose is to count the number of identical words in the input text. (Flink's releases come with this example job)
+
+#### Uploading the main package
+
+When using the Flink task node, you will need to use the Resource Centre to upload the jar package for the executable. Refer to the [resource centre](../resource.md).
```
Comment (zhongjiajie) [inline docs/en-us/2.0.3/user_doc/guide/task/flink.md:51]:
```suggestion
When using the Flink task node, you will need to use the Resource Centre to upload the jar package for the executable. Refer to the [resource center](../resource.md).
```

Comment (zhongjiajie) [reply docs/en-us/2.0.3/user_doc/guide/task/flink.md:51]: find one typo here, same as dev document

Comment (QuakeWang) [reply docs/en-us/2.0.3/user_doc/guide/task/flink.md:51]: > find one typo here, same as dev document

This is probably not a typo, the way it is spelled in British English.

Comment (zhongjiajie) [reply docs/en-us/2.0.3/user_doc/guide/task/flink.md:51]: My bad

Comment (zhongjiajie) [reply docs/en-us/2.0.3/user_doc/guide/task/flink.md:51]: But I find out we use `resource center` in https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/resource.html. I thinks maybe use the same keyword would be better. WDYT?

**Comment by zhongjiajie** [comment]
Oh, we still need to pass CI

**Pull request merged**